gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GitFlowGraphMonitor.java

```diff
@@ -128,11 +131,15 @@ public boolean shouldPollGit() {
    */
   @Override
   void processGitConfigChanges() throws GitAPIException, IOException {
-    if (flowTemplateCatalog.isPresent() && (flowTemplateCatalog.get() instanceof ObservingFSFlowEdgeTemplateCatalog)) {
-      ObservingFSFlowEdgeTemplateCatalog catalog = (ObservingFSFlowEdgeTemplateCatalog) flowTemplateCatalog.get();
-      if (catalog.isShouldRefreshFlowGraph()) {
-        this.gitRepo.initRepository();
-        catalog.setShouldRefreshFlowGraph(false);
+    if (lock.tryLock()) {
```

Comment:
Shouldn't we be using the lock defined in the ObservingFlowCatalog here? Defining a new lock here is not useful since the GitFlowGraphMonitor is single threaded.

Reply:
Hmm okay that's what I was thinking but I'm a bit confused what the concurrency issue is then. I guess the problem is that this variable could be modified from both git monitor and catalog at the same time?

In that case it should work to modify the `setShouldRefreshFlowGraph` in catalog to acquire the same rwlock right?

Reply:
So here is the race condition I was thinking of: 
1. FlowCatalog has been refreshed so shouldRefreshFlowGraph is marked true. 
2. GitMonitoringService calls processGitConfigChanges() and at some point before this method is done, FlowCatalog observes another refresh and sets shouldRefreshFlowGraph flag to true. 
3. This flag is reset by GitMonitoringService to false at the end of processGitConfigChanges(). 

So the GitMonitoringService can miss the 2nd catalog refresh and fail to add edges. LMK, if this is still not clear.

Reply:
Makes sense, updated to use the same lock as the catalog.